apps/pre-award/copilot/environments/addons/fsd-pre-award-stores-cluster.yml

```diff
@@ -0,0 +1,158 @@
+      GenerateSecretString:
+        SecretStringTemplate: '{"username": "postgres"}'
+        GenerateStringKey: "password"
+        ExcludePunctuation: true
+        IncludeSpace: false
+        PasswordLength: 16
+  fsdpreawardstoresclusterDBClusterParameterGroup:
+    Metadata:
+      'aws:copilot:description': 'A DB parameter group for engine configuration values'
+    Type: 'AWS::RDS::DBClusterParameterGroup'
+    Properties:
+      Description: !Ref 'AWS::StackName'
+      Family: 'aurora-postgresql14'
+      Parameters:
+        client_encoding: 'UTF8'
+  
+  fsdpreawardstoresclusterDBCluster:
+    Metadata:
+      'aws:copilot:description': 'The fsdpreawardstorescluster Aurora Serverless v2 database cluster'
+    Type: 'AWS::RDS::DBCluster'
+    Properties:
+      MasterUsername:
+        !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdpreawardstoresclusterAuroraSecret, ":SecretString:username}}" ]]
+      MasterUserPassword:
+        !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdpreawardstoresclusterAuroraSecret, ":SecretString:password}}" ]]
+      DatabaseName: !Ref fsdpreawardstoresclusterDBName
+      Engine: 'aurora-postgresql'
+      EngineVersion: '14.4'
+      DBClusterParameterGroupName: !Ref fsdpreawardstoresclusterDBClusterParameterGroup
+      DBSubnetGroupName: !Ref fsdpreawardstoresclusterDBSubnetGroup
+      Port: 5432
+      VpcSecurityGroupIds:
+        - !Ref fsdpreawardstoresclusterDBClusterSecurityGroup
+      ServerlessV2ScalingConfiguration:
+        # Replace "All" below with "!Ref Env" to set different autoscaling limits per environment.
+        MinCapacity: !FindInMap [fsdpreawardstoresclusterEnvScalingConfigurationMap, All, DBMinCapacity]
+        MaxCapacity: !FindInMap [fsdpreawardstoresclusterEnvScalingConfigurationMap, All, DBMaxCapacity]
+  
+  fsdpreawardstoresclusterDBWriterInstance:
+    Metadata:
+      'aws:copilot:description': 'The fsdpreawardstorescluster Aurora Serverless v2 writer instance'
+    Type: 'AWS::RDS::DBInstance'
+    Properties:
+      DBClusterIdentifier: !Ref fsdpreawardstoresclusterDBCluster
+      DBInstanceClass: db.serverless
+      Engine: 'aurora-postgresql'
+      PromotionTier: 1
+      AvailabilityZone: !Select
+        - 0
+        - !GetAZs
+          Ref: AWS::Region
+
+  fsdpreawardstoresclusterSecretAuroraClusterAttachment:
+    Type: AWS::SecretsManager::SecretTargetAttachment
+    Properties:
+      SecretId: !Ref fsdpreawardstoresclusterAuroraSecret
+      TargetId: !Ref fsdpreawardstoresclusterDBCluster
+      TargetType: AWS::RDS::DBCluster
+
+Outputs:
+  fsdpreawardstoresclusterSecret:
+    Description: "The JSON secret that holds the database username and password. Fields are 'host', 'port', 'dbname', 'username', 'password', 'dbClusterIdentifier' and 'engine'"
+    Value: !Ref fsdpreawardstoresclusterAuroraSecret
+    Export:
+      Name: !Sub ${App}-${Env}-fsdpreawardstoresclusterAuroraSecret
```

Comment:
I think pre-award apps use more like a custom secret structure in place of this?

Reply:
I've updated it to use the same thing as post-award, because pre-award's approach is not secure